server/apps/immich/src/api-v1/asset/asset.service.spec.ts

```diff
@@ -67,29 +113,65 @@ describe('AssetService', () => {
   });
 
   // Currently failing due to calculate checksum from a file
-  // it('create an asset', async () => {
-  //   const assetEntity = _getAsset();
+  it('create an asset', async () => {
+    const assetEntity = _getAsset_1();
 
-  //   assetRepositoryMock.create.mockImplementation(() => Promise.resolve<AssetEntity>(assetEntity));
+    assetRepositoryMock.create.mockImplementation(() => Promise.resolve<AssetEntity>(assetEntity));
 
-  //   const originalPath =
-  //     'upload/3ea54709-e168-42b7-90b0-a0dfe8a7ecbd/original/116766fd-2ef2-52dc-a3ef-149988997291/51c97f95-244f-462d-bdf0-e1dc19913516.jpg';
-  //   const mimeType = 'image/jpeg';
-  //   const createAssetDto = _getCreateAssetDto();
-  //   const result = await sui.createUserAsset(authUser, createAssetDto, originalPath, mimeType);
+    const originalPath = 'fake_path/asset_1.jpeg';
+    const mimeType = 'image/jpeg';
+    const createAssetDto = _getCreateAssetDto();
+    const result = await sui.createUserAsset(
+      authUser,
+      createAssetDto,
+      originalPath,
+      mimeType,
+      Buffer.from('0x5041E6328F7DF8AFF650BEDAED9251897D9A6241', 'hex'),
+    );
 
-  //   expect(result.userId).toEqual(authUser.id);
-  //   expect(result.resizePath).toEqual('');
-  //   expect(result.webpPath).toEqual('');
-  // });
+    expect(result.userId).toEqual(authUser.id);
+    expect(result.resizePath).toEqual('');
+    expect(result.webpPath).toEqual('');
+  });
 
   it('get assets by device id', async () => {
-    assetRepositoryMock.getAllByDeviceId.mockImplementation(() => Promise.resolve<string[]>(['4967046344801']));
+    const assets = _getAssets();
+
+    assetRepositoryMock.getAllByDeviceId.mockImplementation(() =>
+      Promise.resolve<string[]>(Array.from(assets.map((asset) => asset.deviceAssetId))),
```

Comment:
It is tempting to use a loop to get the IDs here. In this case it doesn't seem too bad. However, the general rule is to not use loops or conditions inside the test, makes it more simple and easier to read. Hard-coding the array of `strings` would be preferable.